src/execution_plan/ops/op_delete.c

```diff
@@ -88,15 +68,30 @@ static Record DeleteConsume(OpBase *opBase) {
 	Record r = OpBase_Consume(child);
 	if(!r) return NULL;
 
-	/* Enqueue entities for deletion. */
-	for(int i = 0; i < op->node_count; i++) {
-		Node *n = Record_GetNode(r, op->nodes_to_delete[i]);
-		op->deleted_nodes = array_append(op->deleted_nodes, *n);
-	}
-
-	for(int i = 0; i < op->edge_count; i++) {
-		Edge *e = Record_GetEdge(r, op->edges_to_delete[i]);
-		op->deleted_edges = array_append(op->deleted_edges, *e);
+	/* Expression should be evaluated to either a node or an edge
+	 * which will be marked for deletion, if an expression is evaluated
+	 * to a different value type e.g. Numeric a run-time expection is thrown. */
+	for(int i = 0; i < op->exp_count; i++) {
+		AR_ExpNode *exp = op->exps[i];
+		SIValue value = AR_EXP_Evaluate(exp, r);
+
+		if(!(SI_TYPE(value) & SI_GRAPHENTITY)) {
```

Comment:
This logic could be put in the `else` block just below, but I don't have a strong preference if you think this is cleaner.

Reply:
Absolutely!